pkg/kudoctl/resources/dependencies/resolve.go

```diff
@@ -123,15 +108,42 @@ func dependencyWalk(
 				"failed to resolve package %s, dependency of package %s: %v", fullyQualifiedName(childTask.Spec.KudoOperatorTaskSpec), parent.OperatorVersion.FullyQualifiedName(), err)
 		}
 
+		// after resolving the dependency we update the KudoOperatorTask.Spec definition with the resolved
+		// operator name, version and app version so that a definition like:
+		// ---
+		//   - name: deploy-child
+		//    kind: KudoOperator
+		//    spec:
+		//      package: "../child-operator"
+		//
+		// will be updated as:
+		// ---
+		//   - name: deploy-child
+		//    kind: KudoOperator
+		//    spec:
+		//      appVersion: 3.2.1
+		//      operatorVersion: 0.0.1
+		//      package: child
+		//
+		// so that the KUDO controller to be able to grab the right 'OperatorVersion' resources from the cluster
+		// when the task is executed.
+		err = updateResolvedKudoOperatorTask(childTask.Name,
```

Comment:
See [this comment](https://github.com/kudobuilder/kudo/pull/1728/files#r511488890) on why this logic has been moved here.